libs/wingcompiler/src/compile.ts

```diff
@@ -317,13 +317,20 @@ async function runPreflightCodeInWorkerThread(
 ): Promise<void> {
   try {
     // Create a shimmed entrypoint that ensures we always load the compiler's version of the SDK
+    const sdkEntrypoint = require.resolve("@winglang/sdk");
     const shim = `\
 var Module = require('module');
 var original_resolveFilename = Module._resolveFilename;
-var WINGSDK_PATH = '${normalPath(require.resolve("@winglang/sdk"))}';
+var WINGSDK = '@winglang/sdk';
+var WINGSDK_PATH = '${normalPath(sdkEntrypoint)}';
+var WINGSDK_DIR = '${normalPath(join(sdkEntrypoint, "..", ".."))}';
 
 Module._resolveFilename = function () {
-  if(arguments[0] === '@winglang/sdk') return WINGSDK_PATH;
+  const path = arguments[0];
+  if(path === WINGSDK) return WINGSDK_PATH;
+  if(path.startsWith(WINGSDK)){
+    arguments[0] = path.replace(WINGSDK, WINGSDK_DIR);
+  }
```

Comment:
I love and hate this.

Its like is this "clever as fuck" or "hacky as fuck"... probably someone in the middle "works as fuck" 😁

Reply:
Feels like we might want to advertise this to potential platform builders

Reply:
This is a super hacky version of a [normal thing](https://nodejs.org/api/module.html#customization-hooks) that isn't stable yet

Reply:
Ideally we should use the builtin thing or look into a more overhauled system of dealing with modules. e.g. I've been thinking about integrating something like https://github.com/unjs/jiti